domains/certificates/Query.API/API.IntegrationTests/ContractTests.cs

```diff
@@ -166,6 +167,55 @@ public async Task CreateContract_MeteringPointNotOwnedByUser_BadRequest()
         response.StatusCode.Should().Be(HttpStatusCode.BadRequest);
     }
 
+    [Fact]
+    public async Task CreateContract_WithConsumptionMeteringPoint_TechnologyNull()
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var technology = new Technology(AibFuelCode: "F01040100", AibTechCode: "T010000");
+        dataSyncWireMock.SetupMeteringPointsResponse(gsrn, MeteringPointType.Consumption, technology);
+
+        var subject = Guid.NewGuid().ToString();
+        using var client = factory.CreateAuthenticatedClient(subject);
+
+        var startDate = DateTimeOffset.Now.ToUnixTimeSeconds();
+        var body = new { gsrn, startDate, endDate = (long?)null };
+
+        using var response = await client.PostAsJsonAsync("api/certificates/contracts", body);
+
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var createdContractUri = response.Headers.Location;
+        var createdContract = await client.GetFromJsonAsync<Contract>(createdContractUri);
+
+        createdContract?.Technology?.FuelCode.Should().BeNullOrWhiteSpace();
+        createdContract?.Technology?.TechCode.Should().BeNullOrWhiteSpace();
+    }
+
+    [Fact]
+    public async Task CreateContract_WithProductionMeteringPoint_TechnologyExists()
+    {
+        var gsrn = GsrnHelper.GenerateRandom();
+        var technology = new Technology(AibFuelCode: "F01040100", AibTechCode: "T010000");
+        dataSyncWireMock.SetupMeteringPointsResponse(gsrn, MeteringPointType.Production, technology);
+
+        var subject = Guid.NewGuid().ToString();
+        using var client = factory.CreateAuthenticatedClient(subject);
+
+        var startDate = DateTimeOffset.Now.ToUnixTimeSeconds();
+        var body = new { gsrn, startDate, endDate = (long?)null };
+
+        using var response = await client.PostAsJsonAsync("api/certificates/contracts", body);
+
+        response.StatusCode.Should().Be(HttpStatusCode.Created);
+
+        var createdContractUri = response.Headers.Location;
+        var createdContract = await client.GetFromJsonAsync<Contract>(createdContractUri);
+
+        createdContract!.Technology.Should().NotBeNull();
+        createdContract.Technology!.FuelCode.Should().Be(technology.AibFuelCode);
+        createdContract.Technology.TechCode.Should().Be(technology.AibTechCode);
```

Comment:
Can also be a one-liner, something like

```suggestion
        createdContract!.Technology.Should().Be(technology);
```